### PR TITLE
ci: standardise conventional commits

### PR DESCRIPTION
**Pull request by dabeeeenster**
- Rationalise Conventional Commits to this list added to the docs in this PR
- Auto label PRs based on commit title
- Add docs with approved project labels

### CODE REVIEW

**Comment by matthewelwell** [comment]
@dabeeeenster heads up that I cancelled the vercel deployments to try and jump the ones I need up the queue. This PR doesn't need vercel preview deployments.

File: .github/workflows/platform-pull-request.yml [hunk 1]
```diff
@@ -12,22 +12,40 @@ jobs:
...
         with:
           types: | # mirrors changelog-sections in the /release-please-config.json
-            fix
             feat
+            fix
             infra
+            ci
             docs
-            chore
-            build
```
Comment (zachaysan) [inline .github/workflows/platform-pull-request.yml:25]: Do we not need the `build` designator?

**Review by zachaysan — APPROVED** [review]
Conditionally approved but I'm not sure if we want to discard the `build` category

File: .github/workflows/platform-pull-request.yml (hunk 1 above)
Comment (dabeeeenster) [reply .github/workflows/platform-pull-request.yml:25]: I wanted to reduce the number of labels overall where possible and don't feel its too different to `ci` unless I'm missing something?

File: .github/labeler.yml [hunk 2]
```diff
@@ -15,3 +15,7 @@ api:
 github:
```
Comment (matthewelwell) [inline .github/labeler.yml]: I'd say the `github` label is unclear tbh. We should just use the `ci` label here imo or, if this is a different _category_ of labelling to what we're doing with conv commits [here](https://github.com/Flagsmith/flagsmith/pull/4084/files#diff-88b40629df399d3611837cb436b6d43f743791f1f5944ea002bb76bf3713046aR32), maybe we should use `workflows` instead?

File: docs/docs/platform/contributing.md [hunk 3]
```diff
@@ -16,6 +16,22 @@ suggesting the next steps.
 - If your PR involves a lot of commits, squash them using `git rebase -i` as this makes it easier for us to review.
 - Keep lines under 80 characters.
 
+### Conventional Commits
+
+Please title your PR's with starting with one of the following
```
Comment (matthewelwell) [inline docs/docs/platform/contributing.md]:
```suggestion
Please prefix the title of your PRs with one of the following
```

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: .github/labeler.yml (hunk 2 above)
Comment (dabeeeenster) [reply .github/labeler.yml]: Agreed - had this thought in the shower this morning 🤣

File: .github/workflows/platform-pull-request.yml (hunk 1 above)
Comment (zachaysan) [reply .github/workflows/platform-pull-request.yml:25]: Ok that makes sense.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**